pkg/codegen/codegen.go

```diff
@@ -996,3 +1089,52 @@ func GetParametersImports(params map[string]*openapi3.ParameterRef) (map[string]
 	}
 	return res, nil
 }
+
+func GetPathsImports(paths openapi3.Paths, excludeSchemas []string) (map[string]goImport, error) {
+	res := map[string]goImport{}
+	for _, pathName := range SortedPathsKeys(paths) {
+		for _, method := range SortedOperationsKeys(paths[pathName].Operations()) {
+			operation := paths[pathName].Operations()[method]
+
+			for _, statusCodeOrDefault := range SortedResponsesKeys(operation.Responses) {
+				v := operation.Responses[statusCodeOrDefault]
+				sortedContentKeys := SortedContentKeys(v.Value.Content)
+				for _, mediaTypeStr := range sortedContentKeys {
+					mediaType := v.Value.Content.Get(mediaTypeStr)
+					// fmt.Printf("mediaType.Schema: %v\n", mediaType.Schema)
+					// fmt.Printf("mediaType.Schema: %#v\n", mediaType.Schema)
+					// TODO **************************************************
+					// TODO: this is where things aren't working
+					// TODO **************************************************
+					imprts, err := GoSchemaImports(mediaType.Schema)
```

Comment:
Latest debugging has shown that in here, we're getting an `sref` of:

```
sref: &{#/components/schemas/pascalSchema 0x1400021a1a0}
```

Which is because at the point it's being retrieved, it's true, because it's being retrieved from inside the other spec.

Maybe this means that way that I'm pulling out all the referenced things in `GenerateTypesForPaths` is being _too_ exhaustive, and is reaching into other referenced files, which isn't ideal, as we should only be generating things from the base file